torch/csrc/autograd/saved_variable.cpp

```diff
@@ -22,7 +22,7 @@ SavedVariable::SavedVariable(const Variable& variable, bool is_output) {
     has_grad_fn_ = !variable.is_leaf();
     // These copies are all shared_ptr copies, so slightly more expensive.
     // Do them here instead of in the init list in case data is undefined.
-    data_ = variable.tensor_data();
+    data_ = variable.variable_data();
```

Comment:
doesn't it not matter whether we use `variable_data` or `detach` here, because we stitch up the version_counter later anyway?  If that's the case, I think using `detach` is preferable, since the semantics are clearer.